pkg/sys/vfs/fs.go

```diff
@@ -0,0 +1,100 @@
+/*
+Copyright © 2022 - 2025 SUSE LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vfs
+
+import (
+	"io/fs"
+	"os"
+
+	"github.com/twpayne/go-vfs/v4"
+)
+
+type vfsOS struct {
+	osfs vfs.FS
+}
+
+func OSFS() *vfsOS {
```

Comment:
This highlights the following warning / linter error:
> Exported function with the unexported return type 

Also applies to the other types we initialise.

Reply:
Yes, I prefer to keep interface implementations unexported, this way we can keep implementation and interface entirely separate and we enforce the interface use everywhere else.

I don't see any problem here.